### ISSUE TITLE
Python 3.10 compatibility

### ISSUE THREAD

**Issue by ondrejj** [formencode/formencode#160]
Version check in setup.py is not compatible with python-3.10. Reported in Fedora bugzilla:

https://bugzilla.redhat.com/show_bug.cgi?id=1897581

```
Traceback (most recent call last):
  File "/builddir/build/BUILD/FormEncode-1.3.1/setup.py", line 15, in <module>
    raise ImportError('Python version not supported')
ImportError: Python version not supported
```

Please, use a different version testing, for example:
```
py_version = sys.version_info[0]+sys.version_info[1]/100
if not 2.07 <= py_version < 3.0 and not 3.06 <= py_version:
```

**Comment by hroncok** [comment]
For testing Python versions, use tuples:

```python
if not (2, 7) <= sys.version_info[:2] < (3, 0) and not (3, 6) <= sys.version_info[:2]:
    ...
```

Never floats.

**Comment by ondrejj** [comment]
Can you please make a release? Thank you.

**Comment by lambacck** [comment]
I'm starting with attempting to put 3.10 in the test matrix to see if there are other problems.

**Comment by ondrejj** [comment]
Again please, can you make a release with this fix? Thank you.
https://bugzilla.redhat.com/show_bug.cgi?id=1897581

**Comment by kiilerix** [comment]
With Python 3.10 coming out soon, it is essential to get a new FormEncode release with support for it.

The unfortunate alternative would be to consider it a dead project and invest in moving away from FormEncode.

**Comment by lambacck** [comment]
I have released 2.0.1 to pypi which includes the change to make setup.py work on Python 3.10. The [docs have been updated](https://github.com/formencode/formencode/blob/2.0.1/docs/whatsnew-2.0.txt) to reflect that Python 3.10 support was added but a doc build hasn't run yet on the official doc site. Once the doc build runs the details will show [there](http://www.formencode.org/en/latest/whatsnew-2.0.html)
